### ISSUE TITLE
Incorrect armor values for level 58+ creatures

### ISSUE THREAD

**Issue by T1ti** [vmangos/core#814]
**Current behavior**: 
Creatures level 60 to 63 currently use wrong armor values that are taken from the Bestiary book which is from TBC. 
TBC increased the armor values of all level 60+ mobs so all the known class level templates are wrong for vanilla and need to be updated.

**Expected behavior**: 
Update the class level template armor value of all lvl 60/63 mobs with vanilla values.

Proof of that : https://web.archive.org/web/20060413193842/http://img.photobucket.com/albums/v26/ErikVKing/Magmadar-BeastLore.jpg 
https://www.kurai.com/wow/naxx/maexxna.jpg
And also theorycrafts on classic all found values that exactly match those
https://docs.google.com/spreadsheets/d/1f627xZlZ3x9tRAJwfCNmVMk85v6aTVZhO2_lTDCu0us/edit#gid=442679981

Current values :     
lvl 60 warrior : 3791 armor
lvl 61 warrior : 4091
lvl 60 warrior : 4391
lvl 60 warrior : 4691

Expected values :
lvl 60 warrior : ~3566 Unconfirmed, from theorycrafts
lvl 61 warrior : ~3622 Unconfirmed, from theorycrafts
lvl 60 warrior : ~3677 Unconfirmed, from theorycrafts
lvl 63 warrior : 3731 Confirmed from Beast lore.

We need a hunter on classic using beast lore to verify the armor values of lvl 60/61/62 monsters. 
Most beasts are warriors so finding paladin/mage beasts will be near impossible, but we can calculate them, paladin is 80-81% of warrior, mage is 50-50.5% of warrior. The only non warrior lvl 60+ beast is "The Razza", the arena boss in dire maul.
List of high lvl beasts : https://classicdb.ch/?npcs&filter=type=[1];minl=50;maxl=63;#0-2+1

Veryfing other levels would also be nice but only 60+ are drastically different(also health and dmg ?)
PS : some bosses armor values are even wrong with the bestiary, Thaddius patchwerk and loatheb have 5491 in vmangos but 4691 in bestiary.

**Steps to reproduce**:

1.  Use beast lore on classic with a hunter to compare values with vmangos.

**Client version**: 
1.12

**Comment by kampeador** [comment]
I doubt that only 60-63 lvls are wrong.
Forest Spider in Elwynn Forest.
lvl 5: 126 armor
lvl 6: 180 armor
VMangos db: 147.

Stormpike Owl 54 lvl: 3237 armor
VMangos db: 0

Alterac Ram 51 lvl: 3072 armor
VMangos db: 2200.

Either my DB is old or everything is different in classic.

**Comment by Daribon** [comment]
@kampeador you have to take in mind that vmangos DB currently only has 1 column for armor. So if a mob is supposed to have between 250 and 300 armor on a level 18-20 mob, at both level 18 and 20 it will have 300 armor in vmangos.

Thus far we been comparing a lot with bestiary guide which has accurate data, but with TBC modifiers.
If you don't have bestiary guide, DM me on discord.

**Comment by ratkosrb** [comment]
@kampeador You are alive! I remember you from Elysium times :)

The only creature stats that are accurate all over the board are health and mana. Armor, damage and resistances for most mobs are the same as they were in mangos db in 2011 when Nost forked their project, and who knows whats the source of values there. We can use the Bestiary book to correct stats for important mobs, but nobody has time to go through the whole book and correct all mobs unfortunately.

**Comment by kampeador** [comment]
I've got a bestiary guide from @Daribon and compared it with some mobs in classic using Bestial Lore.
Most of them are +- 50-200 armor, while some are completely different.
For example:
In bestiary and VMangos DB Mistvale Gorilla armor values are listed as 1270-1304, while they have only 54-56 armor in classic.
I'll try to make a table(possibly at the end of this week) so you'll see a difference.

**Comment by T1ti** [comment]
Note : TBC changed the health value of all level 20+ mobs, I expect they done something similar with armor

**Comment by ratkosrb** [comment]
That's not true.

**Comment by Daribon** [comment]
HP value was never changed in TBC. From what I recall the most obvious change in TBC, especially in patch 2.3.0 was a heavy nerf on damage that elite mobs dealt.

**Comment by T1ti** [comment]
Ok sorry I was comparing to the cmangos table which I tought was tbc

**Comment by Daribon** [comment]
> 
> 
> @kampeador You are alive! I remember you from Elysium times :)
> 
> The only creature stats that are accurate all over the board are health and mana. Armor, damage and resistances for most mobs are the same as they were in mangos db in 2011 when Nost forked their project, and who knows whats the source of values there. We can use the Bestiary book to correct stats for important mobs, but nobody has time to go through the whole book and correct all mobs unfortunately.

I think I have an idea where the original values in mangos DB comes from.

Mangos DB was created by forking UDB (Unified Database) sometime in 2008. UDB was created when SDB (Silver-DB) and MoDB merged in September 2007. Prior to this mangos had used mainly SDB for it's database, for example if you checked out those old 2006 repacks. Considering just how many spawns are still shared with UDB, it's safe to say it's very likely that much of the values originally comes from UDB as well.

In original the mangos-zero DB there used to be some big remnants of SDB as well. The horribly spawned Alterac Valley that everyone experienced on the early 2010s private servers, was taken straight from SDB.

**Comment by kampeador** [comment]
No. Mob "health" values a mostly corrent, but it looks like there were some rounding issues when they were generated.
For example, a level 20 Rapid Dire Wolf has 493 hp, while it should have 494.
The higher the level of the creature, the greater the difference in armor. There is also a difference in "health" and "damage".

Here are some mob stats from classic:
https://pastebin.com/Rkh2djjf

**Comment by Daribon** [comment]
What is the source of the above pastebin? images or sniffs?

**Comment by T1ti** [comment]
He said he was just using beast lore on a lot of mobs in classic.
Those values exactly match those from theorycrafts 

![vanillacls](https://user-images.githubusercontent.com/40864460/95619939-b0df3d80-0a6f-11eb-9f43-0918de211c4f.png)

**Comment by T1ti** [comment]
Even for pre lvl 60 the values are slightly off
Exemple : lvl 58 in vmangos is 3435, in classic it is 3457.

I checked and the values from the bestiary are the same as vmangos so both are  wrong.
noxious plaguebat lvl 56 :
-3348 in classic
-3327 in bestiary
-3327 in vmangos

I will make a full list of all lvl with classic values based on that

**Comment by T1ti** [comment]
![armorclassic](https://user-images.githubusercontent.com/40864460/104539452-d7d47f00-561d-11eb-87ca-44797d20a1db.png)
This is the correct armor values for each level from Classic (beast lore) for the Warrior type creatures.

All health values are good except a small mistake for lvl 30

**Comment by NickTyrer** [comment]
@ratkosrb can this be closed now?

**Comment by ratkosrb** [comment]
Have not updated all creatures in the game yet.
